app/src/main/java/ch/epfl/sdp/appart/database/FirestoreDatabaseService.java

```diff
@@ -293,6 +293,19 @@ public FirestoreDatabaseService() {
         return result;
     }
 
+    @NonNull
+    @Override
+    public CompletableFuture<Boolean> putImage(Uri uri, String name, String path) {
+        if (uri == null || name == null) {
+            throw new IllegalArgumentException("parameters cannot be null");
+        }
+        CompletableFuture<Boolean> isFinishedFuture = new CompletableFuture<>();
+        StorageReference storageReference = FirebaseStorage.getInstance().getReference(path); //Firebase Storage Name Folder: example: Ad/0AmzXBySMsOPL9dQ3yKG/photo1
```

Comment:
the storage reference is already available in the class, use the `storage` attribute.